tests/systems/systems_test.C

```diff
@@ -583,6 +728,30 @@ public:
     CPPUNIT_ASSERT_DOUBLES_EQUAL(system.solution->l1_norm(), ref_l1_norm, TOLERANCE*TOLERANCE);
   }
 
+  void testAssemblyWithDgFemContext()
+  {
+    Mesh mesh(*TestCommWorld);
+
+    EquationSystems es(mesh);
+    System &sys = es.add_system<LinearImplicitSystem> ("test");
+
+    // We must use a discontinuous variable type in this test or
+    // else the sparsity pattern will not be correct
+    sys.add_variable("u", FIRST, L2_LAGRANGE);
+
+    MeshTools::Generation::build_cube (mesh,
+                                       5, 5, 5,
+                                       0., 1., 0., 1., 0., 1.,
+                                       HEX8);
+
+    es.init();
+    sys.attach_assemble_function (assembly_with_dg_fem_context);
+    sys.solve();
```

Comment:
I'd like to pull this PR and see how long it takes to run the unit tests in dbg mode. 

Do you think the test would still be sufficient if we just called `sys.assemble()`? Would it be OK if I changed it to 2D?

Reply:
Sure, please proceed as you like. It's fine with me if you change this to 2D.

Reply:
Actually I think I'll just merge as is, the test only increased the dbg mode unit test runtime by about 1% (47.7s -> 48.2s).